ydb/core/cms/api_adapters.cpp

```diff
@@ -212,8 +212,24 @@ class TListClusterNodes: public TAdapterActor<
 
 }; // TListClusterNodes
 
+template <typename TResult>
+class TCompositeActionGroupHandler {
+protected:
+    Ydb::Maintenance::ActionGroupStates* GetActionGroupState(TResult& result) const {
+        if (HasSingleCompositeActionGroup && !result.action_group_states().empty()) {
+            return result.mutable_action_group_states(0);
+        }
+        return result.add_action_group_states();
+    }
+
+protected:
+    bool HasSingleCompositeActionGroup = false;
+};
+
 template <typename TDerived, typename TEvRequest>
-class TPermissionResponseProcessor: public TAdapterActor<TDerived, TEvRequest, TEvCms::TEvMaintenanceTaskResponse> {
+class TPermissionResponseProcessor: 
+    public TAdapterActor<TDerived, TEvRequest, TEvCms::TEvMaintenanceTaskResponse>,
+    public TCompositeActionGroupHandler<Ydb::Maintenance::MaintenanceTaskResult> {
```

Comment:
```suggestion
class TPermissionResponseProcessor
    : public TAdapterActor<TDerived, TEvRequest, TEvCms::TEvMaintenanceTaskResponse>
    , public TCompositeActionGroupHandler<Ydb::Maintenance::MaintenanceTaskResult>
{
```

Благодаря такому форматирование уменьшается дифф при последующих изменениях списка базовых классов.